torch/ao/quantization/fx/prepare.py

```diff
@@ -1205,10 +1208,14 @@ def insert_observers_for_model(
 
     input_quantized_idxs: List[int] = prepare_custom_config.input_quantized_indexes
     output_quantized_idxs: List[int] = prepare_custom_config.output_quantized_indexes
-    processed_nodes: Set[List[Node]] = set()
+    processed_nodes: Set[Node] = set()
     # initalize target_dtype_info
-    DEFAULT_QCONFIG_FOR_TARGET_DTYPE_INFO = \
-        QConfig(activation=PlaceholderObserver.with_args(dtype=torch.float32), weight=PlaceholderObserver.with_args(dtype=torch.float32))
+    DEFAULT_QCONFIG_FOR_TARGET_DTYPE_INFO = (
```

Comment:
should this live on backendconfig instead of here?  Seems like this is saying "configuration for default dtype", and since it's default that points to backendconfig instead of qconfig_mapping?

Reply:
I think this can live in the new "QTagger" API code, this is just temporary

Reply:
I think we chatted before that the programmable API is for user intent.  A default output dtype config doesn't seem like user intent, but rather a default capability which makes sense in a target environment.

So, sounds like it would explicitly not be in something like `QTaggerFutureNameTBD`, and instead be in BackendConfig.  What do you think?

Reply:
> A default output dtype config doesn't seem like user intent

I feel it should be, this is like user doing QConfigMapping.set_global(None)?

Reply:
I think we should avoid using `QTaggerFutureNameTBD` as a catch-all for "all configuration things".  It may be the API exposed to users to programmably adjust intent, but it might be clearer to chat about the underlying components when possible.

For "default dtype", it seems that the value has to depend on the backend.  For example, for common NVIDIA gpu/x86 cpu and PyTorch in general, its fp32.  For some custom DSP that does not support fp32, it may be fp16, or even an integer type.

I think when the user is saying `foo.bar.qconfig=None`, the user intent is "use whatever is default for target environment".  The "what is default in target environment" piece is coming from the backend, not the user.

Reply:
I'd push for `QTaggerFutureNameTBD` to not have any configs or defaults at all in its source code - those should be clearly encapsulated as separate objects and imported.

Reply:
I feel the default is a part of the original unquantized pytorch model? let's say we are quantizing for a DSP that only supports int8, if we start with a floating point pytorch model, can we quantize the model to int8 when user just says QConfigMapping().set_global(None)?

Reply:
> I feel the default is don't quantize the model and the default dtype should just be derived from the original unquantized model?

that makes sense as as well.

I think the confusing thing in the current code is that the default is created inside of a utility function.  Seems like this should either be a configuration (default means default of target backend), or it should read the existing dtype (default means what the input program is doing).  Neither of those things should care about `prepare.py`.

Reply:
I think the current state is that it is trying to do the latter, we can certainly improvement this by either just use "None" or read from node.meta["val"] in the future, this PR is just a refactor so I hope to do these fixes in the future

Reply:
I think doing this here is pretty confusing to non-experts as it changes the role of QConfig from "comes from user or default" to "can be created inside code", can we add a comment on the future plan and fix it soon?

Reply:
or how about just

```
// qconfig.py
default_fp32_placeholder_qconfig = ...

// this file can just reference that object
```

Reply:
sure, can I add a private one: `_default_fp32_placeholder_qconfig`? we may delete this soon

Reply:
@vkuzo updated, please take a look again